src/display.hpp

```diff
@@ -975,6 +976,49 @@ class display : public gui2::top_level_drawable
 
 protected:
 	static display * singleton_;
+
+public:
+	static bool add_locked_location(const map_location* mlptr)
```

Comment:
The return value seems to be ignored, so why must this return a value at all?

Creating a RAII class that handles the `remove_locked_location` in its destructor would be better - the design of temporary_unit_mover shows a RAII class.